### PR TITLE
hardening/1052_change_return_creation_of_jsonpayload

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1052 
* 100% Unit tests passed: 
```
Results : Tests run: 66, Failures: 0, Errors: 0, Skipped: 0
```
* Tests: 
```
Running com.telefonica.iot.cygnus.backends.http.HttpBackendTest
[HttpBackend] ---------------------------- Testing HttpBackend.doRequest - 'Gets a valid JSONArray'
[HttpBackend] ----------------------------  OK  - Succesfully got
[HttpBackend] ---------------------------- Testing HttpBackend.doRequest - 'Gets a valid JSONResponse'
[HttpBackend] ----------------------------  OK  - Succesfully got
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 sec
```
* Assignee: @frbattid

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -16,3 +16,4 @@
 - [cygnus-ngsi] [feature] Add MongoDB and STH sinks to docker agent (#1044)
 - [cygnus-ngsi] [hardening] Remove support to deprecated 'matching_table' parameter (#1048)
 - [cygnus-ngsi] [hardening] Add checks for invalid configuration in NGSIGroupingInterceptor.java (#1049)
+- [cygnus-common] [hardening] Change the creation of the jsonpayload of the response in HttpBackend (#1052)
```
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: jsonpayload --> Json payload

Comment (pcoello25) [inline CHANGES_NEXT_RELEASE]: Fixed in 868eef3

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    private class HttpBackendImpl extends HttpBackend {
+
+        public HttpBackendImpl(String[] hosts, String port, boolean ssl, boolean krb5, String krb5User, String krb5Password, String krb5LoginConfFile, String krb5ConfFile) {
+            super(hosts, port, ssl, krb5, krb5User, krb5Password, krb5LoginConfFile, krb5ConfFile);
+        } // HttpBackendImpl
+   
+    } // HttpBackendImpl
+    
+    @Mock
+    HttpClient httpclient; // TBD create setHttpClient en httpBackend
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Leftover comment? Or is it a real need?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java [hunk 3]
```diff
@@ -149,7 +150,7 @@ public JsonResponse doRequest(String method, String url, boolean relative, Array
         } // if else
     } // doRequest
         
-    private JsonResponse doRequest(String method, String url, ArrayList<Header> headers, StringEntity entity)
+    protected JsonResponse doRequest(String method, String url, ArrayList<Header> headers, StringEntity entity)
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java]: Javadoc is required. Something saying what the methods does, and why it is protected (it is protected becuase it is used by the tests).

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        when(mockResponse.getEntity()).thenReturn(normalEntity);
+        
+        when(mockArrayResponse.getEntity()).thenReturn(arrayEntity);
+        when(httpclient.execute(mockRequest)).thenReturn(mockResponse);
+        when(httpclient.execute(mockArrayRequest)).thenReturn(mockArrayResponse);
+    } // setUp
+    
+    @Test
+    public void testDoRequestWithNormalResponse() throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend]") + "- Testing HttpBackend.doRequest - 'Gets a valid JSONResponse'");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - Gets a valid Json object based JSONResponse");

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[HttpBackend]") + " - FAIL - There was some problem when handling the request.");
+            throw e;
+        } // try catch
+        
+    } // testDoRequestWithNormalResponse
+    
+    @Test
+    public void testDoRequestWithArrayResponse () throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend]") + "- Testing HttpBackend.doRequest - 'Gets a valid JSONArray'");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - Gets a valid Json array-based JSONResponse");

Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 4ac3bf2

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 4 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 4ac3bf2

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 2 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: My fault. It was a tip for me :smile: 
Fixed in 4ac3bf2

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java]: Fixed in e0660bc

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    
+    @Test
+    public void testDoRequestWithNormalResponse() throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - Gets a valid Json object based JSONResponse");
+        httpBackend = new HttpBackendImpl(host, normalURL, false, false, null, null, null, null);
+        httpBackend.setHttpClient(httpclient);
+        
+        try {
+            httpBackend.doRequest("GET", normalURL, headers, normalEntity);
+            System.out.println(getTestTraceHead("[HttpBackend]") + "-  OK  - Succesfully got");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: HttpBackend.doRequest

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[HttpBackend]") + " - FAIL - There was some problem when handling the request.");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: HttpBackend.doRequest

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    
+    @Test
+    public void testDoRequestWithArrayResponse () throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - 'Gets a valid Json array based JSONResponse'");
+        httpBackend = new HttpBackendImpl(host, arrayURL, false, false, null, null, null, null);
+        httpBackend.setHttpClient(httpclient);
+        
+        try {
+            httpBackend.doRequest("GET", arrayURL, headers, arrayEntity);
+            System.out.println(getTestTraceHead("[HttpBackend]") + "-  OK  - Succesfully got");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: HttpBackend.doRequest

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[HttpBackend]") + " - FAIL - There was some problem when handling the request.");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: HttpBackend.doRequest

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 8 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 286e241

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 7 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 286e241

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 6 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 286e241

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 9 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in 286e241

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    
+    @Test
+    public void testDoRequestWithArrayResponse () throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - 'Gets a valid Json array based JSONResponse'");
+        httpBackend = new HttpBackendImpl(host, arrayURL, false, false, null, null, null, null);
+        httpBackend.setHttpClient(httpclient);
+        
+        try {
+            httpBackend.doRequest("GET", arrayURL, headers, arrayEntity);
+            System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + "-  OK  - Succesfully got");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: White space before `"-  OK...`

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+    
+    @Test
+    public void testDoRequestWithNormalResponse() throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - Gets a valid Json object based JSONResponse");
+        httpBackend = new HttpBackendImpl(host, normalURL, false, false, null, null, null, null);
+        httpBackend.setHttpClient(httpclient);
+        
+        try {
+            httpBackend.doRequest("GET", normalURL, headers, normalEntity);
+            System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + "-  OK  - Succesfully got");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: White space before `- OK...`

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - FAIL - There was some problem when handling the request.");
+            throw e;
+        } // try catch
+        
+    } // testDoRequestWithNormalResponse
+    
+    @Test
+    public void testDoRequestWithArrayResponse () throws Exception {
+        System.out.println(getTestTraceHead("[HttpBackend.doRequest]") + " - 'Gets a valid Json array based JSONResponse'");
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Simple quotes are not necessary

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 11 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in ee74194

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 10 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in ee74194

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java (hunk 12 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/http/HttpBackendTest.java]: Fixed in f11ed5d

**Comment by pcoello25** [comment]
Finally, you can proceed.

**Comment by frbattid** [comment]
LGTM

**Pull request merged**